modules/tfm/tfm/boards/common/nrf_provisioning.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "tfm_plat_provisioning.h"
+#include "tfm_plat_otp.h"
+#include "tfm_hal_platform.h"
+#include "tfm_attest_hal.h"
+#include "hw_unique_key.h"
+#include "nrfx_nvmc.h"
+#include <nrf_cc3xx_platform.h>
+#include <nrf_cc3xx_platform_identity_key.h>
+#include "nrf_provisioning.h"
+
+int tfm_plat_provisioning_is_required(void)
+{
+    enum tfm_security_lifecycle_t lcs = tfm_attest_hal_get_security_lifecycle();
+
+    return lcs == TFM_SLC_PSA_ROT_PROVISIONING;
+}
+
+enum tfm_plat_err_t tfm_plat_provisioning_perform(void)
+{
+    enum tfm_security_lifecycle_t lcs = tfm_attest_hal_get_security_lifecycle();
+
+    /*
+     * Provisioning in NRF defines has two steps, the first step is to execute
+     * the identity_key_generation sample. When this sample is executed it will
+     * always set the lifecycle state to PROVISIONING. This is a requirement for
+     * the TF-M provisioning to be completed so we don't accept any other
+     * lifecycle state here.
+     */
+
+    /* The Hardware Unique Keys should be already written */
+    if (!hw_unique_key_are_any_written()) {
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    /* The Initial Attestation key should be already written */
+    if (!nrf_cc3xx_platform_identity_key_is_stored(NRF_KMU_SLOT_KIDENT)) {
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    /*
+     * We don't need to make sure that the validation key is written here since we expect
```

Comment:
```suggestion
     * We don't need to make sure that the validation key is written here since we assume
```
By what is this assumption backed up? K/CMake-config so this code shouldn't be reached when SECURE_BOOT isn't enabled?

Reply:
That's a good point, I will add this. Right now if you enable provisioning without secure boot it will fail in runtime. It will be better to fail in Kconfig I guess.